### PR TITLE
Add client statistics.

### PR DESCRIPTION
**Pull request by cezarfx**
Stats are collected on the client application, when system property
-Dcom.oracle.nosql.sdk.nosqldriver.stats.profile=[regular|full] is used,
for an interval of time, set by
-Dcom.oracle.nosql.sdk.nosqldriver.stats.interval=600 in seconds.

The following stats are collected for each type of request: count,
wire latencies, request and response sizes, errors, retries, retry delay,
rate limit delay. Also if rateLimiting is enabled and concurrent connections
number.

StatsConfig also provides an API to programmatically configure the statistics
collection, to selectively start and stop collection and get access to
collected statistics.

### CODE REVIEW

File: driver/src/main/java/oracle/nosql/driver/NoSQLHandle.java [hunk 1]
```diff
@@ -571,6 +571,15 @@ public SystemResult doSystemRequest(String statement,
                                         int timeoutMs,
                                         int pollIntervalMs);
 
+    /**
+     * Returns configuration object for client side statistics.
```
Comment (gmfeinberg) [inline driver/src/main/java/oracle/nosql/driver/NoSQLHandle.java:575]: "client side" seems to be redundant in user-facing material. It's useful when we talk about it internally but for an SDK user I think it should be simply "statistics"

**Comment by gmfeinberg** [comment]
I still have some general comments:
1. The relationship among StatsConfig, StatsConfigImpl, Stats and how a user gets/uses/configures stats is confusing. Usually in our system a "Config" class is created essentially as a complex options object to construction of another class, in this case Stats. This is not the case here. StatsConfig is an interface, StatsConfigImpl is a "live" implementation in that changes to it at run time will (I think) affect the running stats. This is confusing and not typical. I would expect a StatsConfig object (not interface) that is used to create Stats and is not live after that. Ultimately we may want direct user control over starting/stopping/acquiring stats and how that might happen is not clear either. Maybe I'm missing something from the spec

2. I suggested separately to create a getName() interface on Request to avoid the calls like request.getClass().getSimpleName(). Is there a reason to not do this? It'd be simpler and more efficient. Minimally create a local method to do this (String getName(Request)). This approach would also make the mapping of request names more reliable than code like this, which is just not pretty at all:
 `     String requestClass = kvRequest.getClass().getSimpleName();
        requestClass = requestClass.endsWith("Request")
            ? requestClass.substring(0, requestClass.length() - 7) :
            requestClass;

        rStat = requests.get(requestClass);
`        
3. New files are missing Copyright headers

4. Stats.java has no comments at all

**Comment by connelly38** [comment]
> 
>     1. The relationship among StatsConfig, StatsConfigImpl, Stats and how a user gets/uses/configures stats is confusing. Usually in our system a "Config" class is created essentially as a complex options object to construction of another class, in this case Stats. This is not the case here. StatsConfig is an interface, StatsConfigImpl is a "live" implementation in that changes to it at run time will (I think) affect the running stats. This is confusing and not typical. I would expect a StatsConfig object (not interface) that is used to create Stats and is not live after that. Ultimately we may want direct user control over starting/stopping/acquiring stats and how that might happen is not clear either. Maybe I'm missing something from the spec
> 
After reading this again, I agree with George. StatsConfig should just be a class. No need for an Impl.

**Comment by gmfeinberg** [comment]
I'm also confused about what the extra query stats are storing. Can you talk about them? I didn't expect stats per statement that would include the statement and driver side query plan

**Pull request merged**